x-pack/test/api_integration/apis/metrics_ui/sources.ts

```diff
@@ -103,28 +105,65 @@ export default function ({ getService }: FtrProviderContext) {
       it('validates anomalyThreshold is between range 1-100', async () => {
         // create config with bad request
         await supertest
-          .patch(SOURCE_API_URL)
+          .patch(`${SOURCE_API_URL}/${SOURCE_ID}`)
           .set('kbn-xsrf', 'xxx')
           .send({ name: 'NAME', anomalyThreshold: -20 })
           .expect(400);
         // create config with good request
         await supertest
-          .patch(SOURCE_API_URL)
+          .patch(`${SOURCE_API_URL}/${SOURCE_ID}`)
           .set('kbn-xsrf', 'xxx')
           .send({ name: 'NAME', anomalyThreshold: 20 })
           .expect(200);
 
         await supertest
-          .patch(SOURCE_API_URL)
+          .patch(`${SOURCE_API_URL}/${SOURCE_ID}`)
           .set('kbn-xsrf', 'xxx')
           .send({ anomalyThreshold: -2 })
           .expect(400);
         await supertest
-          .patch(SOURCE_API_URL)
+          .patch(`${SOURCE_API_URL}/${SOURCE_ID}`)
           .set('kbn-xsrf', 'xxx')
           .send({ anomalyThreshold: 101 })
           .expect(400);
       });
     });
+
+    describe('has data', () => {
+      const makeRequest = async (params?: {
+        modules?: string[];
+        expectedHttpStatusCode?: number;
+      }) => {
+        const { modules, expectedHttpStatusCode = 200 } = params ?? {};
+        return supertest
+          .get(`${SOURCE_API_URL}/hasData`)
+          .query(modules ? { modules } : '')
+          .set('kbn-xsrf', 'xxx')
+          .expect(expectedHttpStatusCode);
+      };
+
+      before(() => patchRequest({ name: 'default', metricAlias: 'metrics-*,metricbeat-*' }));
```

Comment:
I've seen much code in our repo that does this syntax, dropping the async / await.  
Is there no worry of a timing issue?

Reply:
it returns a promise, jest should recognize it as an async operation a treat it accordingly

Reply:
Indeed, thanks for sharing your perspective.

Reply:
One question, can you tell me which Jest config this test is apart of?